SofaKernel/modules/SofaDefaultType/src/sofa/defaulttype/Vec.h

```diff
@@ -476,7 +476,7 @@ class Vec : public helper::fixed_array<real,N>
     template<class real2>
     Vec<N,real> mulscalar(real2 f) const
     {
-        static_assert(DataTypeInfo<real2>::ValidInfo && DataTypeInfo<real2>::Size==1, "");
+        //static_assert(DataTypeInfo<real2>::ValidInfo && DataTypeInfo<real2>::Size==1, "");
```

Comment:
Est-ce qu'on ne pourrait pas simplement remplacer par

```
static_assert(std::is_arithmetic_v<real2>, "The template argument must be a scalar (i.e. an integer or a floating point type).")
```

Reply:
Yes, I was too lazy to refactor Vec.h as it was out of the scope of my PR. But commenting the code rise a bell in me saying that using moder typetraits feature could helps here.

Reply:
I think removing unnecessarily DataTypeInfo usage found in this PR could remain in this scope (looks like there aren't many in the files you have modified).